pkg/oci/build.go

```diff
@@ -335,3 +356,80 @@ func createImageIndex(ctx context.Context, target oras.Target, o *BuildGadgetIma
 	}
 	return indexDesc, nil
 }
+
+func createManifestForHostTarget(ctx context.Context, target oras.Target, arch, layerPath string) (ocispec.Descriptor, error) {
+	layerDescs := []ocispec.Descriptor{}
+
+	binaryDesc, err := createLayerDesc(ctx, target, layerPath, binaryMediaType)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("creating and pushing binary descriptor: %w", err)
+	}
+	layerDescs = append(layerDescs, binaryDesc)
+
+	configDesc, err := createConfigHostDesc(ctx, target, arch, binaryDesc)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("creating empty descriptor: %w", err)
+	}
+
+	manifest := ocispec.Manifest{
+		Versioned: specs.Versioned{
+			SchemaVersion: 2, // historical value. does not pertain to OCI or docker version
+		},
+		Config: configDesc,
+		Layers: layerDescs,
+	}
+
+	manifestJson, err := json.Marshal(manifest)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("marshalling manifest: %w", err)
+	}
+	manifestDesc := content.NewDescriptorFromBytes(ocispec.MediaTypeImageManifest, manifestJson)
+	manifestDesc.Platform = &ocispec.Platform{
+		Architecture: arch,
+		OS:           "linux",
```

Comment:
Those two are different. This is the manifest that is intended to be run by docker, while the manifest above is intended to contain the gadget ebpf program. I had to use "ebpf" for the gadget otherwise docker will try to run it fail.

I'll investigate this a bit more.